TLM/TMPE.CitiesGameBridge/Service/NetService.cs

```diff
@@ -102,71 +102,108 @@ public class NetService : INetService {
             IterateNodeSegments(nodeId, ClockDirection.None, handler);
         }
 
-        public void IterateNodeSegments(ushort nodeId,
-                                        ClockDirection dir,
-                                        NetSegmentHandler handler) {
-            NetManager netManager = Singleton<NetManager>.instance;
+        /// <summary>
+        /// Given a <paramref name="nodeId"/>, retrieves the assocaited <see cref="NetNode"/>.
+        ///
+        /// A <c>nodeId</c> of <c>0</c> returns <c>false</c>.
+        /// </summary>
+        /// 
+        /// <param name="nodeId">The id of the segment to retrieve.</param>
+        /// <param name="node">The returned segment.</param>
+        /// 
+        /// <returns>Returns <c>true</c> if successful, otherwise <c>false</c>.</returns>
+        public bool TryGetNode(ushort nodeId, out NetNode node) {
```

Comment:
`out` for struct type will perform a copy of `NetNode`

Reply:
Even when it's inlined by the jitter?

Reply:
Double check maybe look at the generated bytecode, we do not want a copy for that. Unless it is called few times and doesn't create any impact.

Reply:
Yeah, I would use ref instead, even it does not communicate the intent as clear as out.

Reply:
ok .. I double checked, looking into the docs and doing my own tests.
out is passing per reference, so the out won't do a copy of NetNode.
but, assigning to node inside TryGetNode will do.
BUT... we are talking about some structs copies versus that closure monster on the left...
its a huge improvement, no matter what you do in that TryGetNode method xD

Reply:
>  but, assigning to node inside TryGetNode will do.

Don't understand what this means.

So just so we are on the same page?
If I modify field members of node outputed by TryGetNode(), the original node structure changes. Am I right?
```C#
TryGetNode(nodeId,out var node);
node.elevation=10;// does this change original node?
```

if you have time it would be nice to put the decompiled code here so that we can check :)

Reply:
@kianzarrin sry, forgot about that one. what I meant is that inside of a method with an out param the compiler forces you to assign that out variable. this is the place where the copy will happen for a struct. but the returning of that copy into the outer scope is done by ref. so there is only one copy, not two. but I could be wrong and have currently no pc, so can't check xD